### ISSUE TITLE
Revise update of scraper targets

### ISSUE THREAD

**Issue by leodido** [influxdata/influxdb#10797]
Currently the patch/update process for scraper targets does not use any struct to specify which fields can be updated, but uses the `platform.ScraperTarget` directly.

Thus I propose to introduce a `platform.ScraperTargetUpdate` struct and use another private struct to decode into it during `PATCH`.

**Comment by desa** [comment]
@leodido what's the status of this issue. I've lost context after the migration.

**Comment by leodido** [comment]
Currently the mechanism we update scraper targets is not homogeneous with other entities.

Thus I proposed to introduce a struct `platform.ScraperTargetUpdate` responsible of specifying which fields can be updated.

**Comment by kelwang** [comment]
Yeah, we can call it ScraperTargetRequest. So it could also match the swagger doc
